File: broker/src/main/scala/io/octopus/scala/broker/mqtt/server/transport/ITransport.scala
package io.octopus.scala.broker.mqtt.server.transport

import io.netty.channel.EventLoopGroup
import io.netty.channel.socket.ServerSocketChannel
import io.octopus.broker.security.ReadWriteControl
import io.octopus.kernel.kernel.config.IConfig
import io.octopus.kernel.kernel.interceptor.NotifyInterceptor
import io.octopus.kernel.kernel.security.IAuthenticator
import io.octopus.kernel.kernel.session.ISessionResistor
import io.octopus.kernel.kernel.subscriptions.ISubscriptionsDirectory
import io.octopus.scala.broker.mqtt.server.PostOffice

/**
 * @author chenxu
 * @version 1
 */

trait ITransport {

  /**
   * 初始化协议
   *
   * @param bossGroup bossGroup
   * @param workerGroup workerGroup
   * @param config config
   * @param sessionRegistry sessionRegistry
   * @param subscriptionsDirectory subscriptionsDirectory
   * @param msgDispatcher msgDispatcher
   * @param ports ports
   * @param authenticator authenticator
   * @param interceptor interceptor
   * @param readWriteControl readWriteControl
   * @param msgQueue msgQueue
   */
  def initProtocol(bossGroup: EventLoopGroup, workerGroup: EventLoopGroup, channelClass: Class[_ <: ServerSocketChannel],
                   config: IConfig, sessionRegistry: ISessionResistor,
                   subscriptionsDirectory: ISubscriptionsDirectory,
                   msgDispatcher: PostOffice,
                   ports: Map[String, Int], authenticator: IAuthenticator,
                   interceptor: NotifyInterceptor, readWriteControl: ReadWriteControl): Unit

}

File: broker-kernel/src/main/java/io/octopus/kernel/kernel/transport/ITransport.java
package io.octopus.kernel.kernel.transport;

import io.netty.channel.EventLoopGroup;
import io.netty.channel.socket.ServerSocketChannel;
import io.octopus.kernel.kernel.config.IConfig;
import io.octopus.kernel.kernel.interceptor.NotifyInterceptor;
import io.octopus.kernel.kernel.postoffice.IPostOffice;
import io.octopus.kernel.kernel.security.IAuthenticator;
import io.octopus.kernel.kernel.security.ReadWriteControl;
import io.octopus.kernel.kernel.session.ISessionResistor;
import io.octopus.kernel.kernel.subscriptions.ISubscriptionsDirectory;

import java.util.Map;

/**
 * @author chenxu
 * @version 1
 * @date 2022/7/12 16:46
 */
public interface ITransport {

    /**
     * 初始化协议
     *
     * @param bossGroup              bossGroup
     * @param workerGroup            workerGroup
     * @param config                 config
     * @param sessionRegistry        sessionRegistry
     * @param subscriptionsDirectory subscriptionsDirectory
     * @param msgDispatcher          msgDispatcher
     * @param ports                  ports
     * @param authenticator          authenticator
     * @param interceptor            interceptor
     * @param readWriteControl       readWriteControl
     */
    void initProtocol(final EventLoopGroup bossGroup, final EventLoopGroup workerGroup, final Class<? extends ServerSocketChannel> channelClass,
                      final IConfig config, final ISessionResistor sessionRegistry,
                      final ISubscriptionsDirectory subscriptionsDirectory,
                      final IPostOffice msgDispatcher,
                      final Map<String, Integer> ports, final IAuthenticator authenticator,
                      final NotifyInterceptor interceptor, final ReadWriteControl readWriteControl);
}